django/contrib/admin/tests.py

```diff
@@ -97,14 +99,34 @@ def wait_until_invisible(self, css_selector, timeout=10):
             timeout
         )
 
-    def wait_page_loaded(self):
+    def wait_page_ready(self, timeout=10):
+        """
+        Block until page has started to load.
+        """
+        self.wait_until(
+            lambda driver: driver.execute_script('return document.readyState;') == 'complete',
+            timeout
+        )
+
+    @contextmanager
+    def wait_page_loaded(self, timeout=10):
         """
         Block until page has started to load.
         """
         from selenium.common.exceptions import TimeoutException
+        from selenium.webdriver.support import expected_conditions as ec
+
         try:
+            old_page = self.selenium.find_element_by_tag_name('html')
+            yield
             # Wait for the next page to be loaded
-            self.wait_for('body')
+            self.wait_until(ec.staleness_of(old_page), timeout=timeout)
+            self.wait_until(
```

Comment:
This is just `wait_page_ready()` no?

Reply:
Good point, that was just natural code progression. I wrote one before the other and didn't refactor at the end 🤷‍♂